src/world.rs

```diff
@@ -0,0 +1,214 @@
+    pub width: Option<u32>,
+    /// The optional height of the map.
+    pub height: Option<u32>,
+}
+
+/// A WorldPattern defines a regex pattern to automatically determine which maps to load and how to lay them out.
+#[derive(Deserialize, PartialEq, Clone, Debug)]
+struct WorldPattern {
+    /// The regex pattern to match against filenames. The first two capture groups should be the x integer and y integer positions.
+    pub regexp: String,
+    /// The multiplier for the x position.
+    #[serde(rename = "multiplierX")]
+    pub multiplier_x: u32,
+    /// The multiplier for the y position.
+    #[serde(rename = "multiplierY")]
+    pub multiplier_y: u32,
+    /// The offset for the x position.
+    #[serde(rename = "offsetX")]
+    pub offset_x: i32,
+    /// The offset for the y position.
+    #[serde(rename = "offsetY")]
+    pub offset_y: i32,
+}
+
+/// Parse a Tiled World file from a path.
+/// If a the Patterns field is present, it will attempt to build the maps list based on the regex patterns.
+///
+/// ## Parameters
+/// - `world_path`: The path to the world file.
+/// 
+/// ## Example
+/// ```
+/// # use tiled::Loader;
+/// #
+/// # fn main() {
+/// #    let mut loader = Loader::new();
+/// #    let world = loader.load_world("assets/world/world_basic.world").unwrap();
+/// #    
+/// #    for map in world.maps.unwrap() {
+/// #        println!("Map: {:?}", map);
+/// #    }
+/// # }
+/// ```
+pub(crate) fn parse_world(
+    world_path: &Path,
+    reader: &mut impl ResourceReader,
+    cache: &mut impl ResourceCache,
+) -> Result<World, Error> {
+    let mut path = reader.read_from(&world_path).map_err(|err| Error::ResourceLoadingError {
+        path: world_path.to_owned(),
+        err: Box::new(err),
+    })?;
+
+    let mut world_string = String::new();
+    path.read_to_string(&mut world_string).map_err(|err| Error::ResourceLoadingError {
+        path: world_path.to_owned(),
+        err: Box::new(err),
+    })?;
+
+    let mut world: World = match serde_json::from_str(&world_string) {
+        Ok(world) => world,
+        Err(err) => {
+            return Err(Error::JsonDecodingError(err));
+        }
+    };
+
+    if world.patterns.is_some() {
+        world.maps = match parse_world_pattern(world_path, &world.clone().patterns.unwrap()) {
+            Ok(maps) => Some(maps),
+            Err(err) => return Err(err),
+        };
+    }
+
+    Ok(world)
+}
+
+/// If "patterns" key is present, it will attempt to build the maps list based on the regex patterns.
+fn parse_world_pattern(path: &Path, patterns: &Vec<WorldPattern>) -> Result<Vec<WorldMap>, Error> {
+    let mut maps = Vec::new();
+
+    let parent_dir = path.parent().ok_or(Error::ResourceLoadingError {
+        path: path.to_owned(),
+        err: Box::new(std::io::Error::from(std::io::ErrorKind::NotFound)),
+    })?;
+
+    // There's no documentation on why "patterns" is a JSON array, so we'll just blast them into same maps list.
```

Comment:
There are multiple patterns because one world can contain sets of maps at different offsets and/or with different naming patterns. It's not common, but I had to support at least one such a case. :-)

Reply:
Latest commit handles multiple patterns and resolves the first matched pattern in the array that matches. I believe this is how Tiled handles it as well.